source/user-portal/assets/favRecipe.js

```diff
@@ -104,5 +125,28 @@ class FavRecipe extends HTMLElement {
 
     }
 }
+
+/**
+ * CREDITS to Lab 6 Starter Code.
+ * 
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+ function searchForKey(object, key) {
+    var value;
+    Object.keys(object).some(function (k) {
+      if (k === key) {
+        value = object[k];
+        return true;
+      }
+      if (object[k] && typeof object[k] === 'object') {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
      if (object[k] && typeof object[k] === "object") {
```